### ISSUE TITLE
Dropping EOL Python Support + Dependency Upgrades

### ISSUE THREAD

**Issue by danclaytondev** [taskipy/taskipy#58]
I wanted to start a conversation regarding continued support for old Python versions, with a view to upgrade taskipy's dependency's to newer versions.

Python 3.6 reached end of life over a year ago and Python 3.7 will reach EOL in June 2023, with no security updates.

In my opinion it makes sense to drop both 3.6 and 3.7 to allow updates with newer package version and Python features. Some dependencies could be removed entirely. Any users who need to use 3.6/3.7 can use an existing version of taskipy, but continued support will make maintaining taskipy more difficult.

Upgrades that can make if supporting Python >= 3.8:
- Mypy v1 is now released and only supports Python >= 3.7.
- New pylint versions
- Colorama

Would it be useful to add a list of tasks that could be contributed to keep taskipy maintained? I would like to contribute if possible! taskipy is exactly what we need for some projects.

**Comment by eugenetriguba** [comment]
Hi @danclaytondev!

> In my opinion it makes sense to drop both 3.6 and 3.7 to allow updates with newer package version and Python features.

Is there a particular reason not to do something similar like what is being done with the `tomli` dependency?

i.e.

```
tomli = [
    { version = "^2.0.1", python = "^3.7" },
    { version = "^1.2.3", python = "~3.6" }
]
```

Something being EOL doesn't mean it isn't still used by the community. If supporting it right now isn't much of a hassle and would benefit a good portion of users still, is there a good reason not to?

**Comment by danclaytondev** [comment]
You're right that it isn't much hassle, but it is still some hassle for maintainers. I don't think people should be encouraged to keep using EOL Python versions with no security updates etc. Python 2 stayed around too long for those reasons. 

If they're stuck with legacy Python then they can use always existing versions of packages but won't get the benefit of new features. If they want new features or bug fixes then they should upgrade to a Python version with official support. 

I can see arguments for both sides but my opinion is to keep new package releases with only supported Python versions. As long as old package releases remain available no one is getting anything worse than they currently have :)

**Comment by illBeRoy** [comment]
IMO this can be a great idea for the V2 milestone (as well as the now ancient request to output errors to stderr). We can decide on a set of features that we want to close the current version with, and then move to V2 that requires 3.8+.

WDYT @eugenetriguba ?

**Comment by eugenetriguba** [comment]
This has been merged into a 2.0 branch and will be in the next major release 🙂
